app/uk/gov/hmrc/bindingtariffclassification/repository/SequenceRepository.scala

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.bindingtariffclassification.repository
+
+import com.google.inject.ImplementedBy
+import javax.inject.{Inject, Singleton}
+import play.api.libs.json.Json
+import reactivemongo.bson.BSONObjectID
+import reactivemongo.play.json.collection.JSONCollection
+import uk.gov.hmrc.bindingtariffclassification.model.JsonFormatters.formatSequence
+import uk.gov.hmrc.bindingtariffclassification.model.{JsonFormatters, Sequence}
+import uk.gov.hmrc.mongo.ReactiveRepository
+import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.Future
+
+@ImplementedBy(classOf[SequenceMongoRepository])
+trait SequenceRepository {
+
+  def insert(e: Sequence): Future[Sequence]
+
+  def getByName(name: String): Future[Sequence]
+
+  def incrementAndGetByName(name: String): Future[Sequence]
+}
+
+@Singleton
+class SequenceMongoRepository @Inject()(mongoDbProvider: MongoDbProvider)
```

Comment:
Do you really need a new collection?
How about taking the max number of reference (in the case repository) and increment it?!

Reply:
Theyre currently stored as strings, It would require quite a bit more work this way to clear all existing data etc

Reply:
Keep it separated for two reasons. **Test in isolation** which makes live much easier and **Separation of concerns** (we could move this logic into a separate service in a separate db)